### DISCUSSION TITLE
How to set filter for direct GO term annotations search?

### DISCUSSION THREAD

**Discussion by jasperhyp** [tanghaibao/goatools#245] [Q&A]
Hi,  I am directly quering annotations of GO terms.  Meanwhile, I also want to set filters, e.g. exclude "NOT" ones.  How can I achieve this?  

My current code looks lilke:
```
    objanno = Gene2GoReader(DATA_DIR+'gene_ontology/gene2go', taxids=[9606])
    go2geneids_human = objanno.get_id2gos(namespace='BP', go2geneids=True, ev_include={'EXP','IDA','IPI','IMP','IGI','IEP'})  # only include experimental evidence, see https://github.com/tanghaibao/goatools/issues/119 and https://github.com/tanghaibao/goatools/blob/main/goatools/anno/opts.py

    print("{N:} GO terms associated with human NCBI Entrez GeneIDs".format(N=len(go2geneids_human)))
    srchhelp = GoSearch(DATA_DIR+"gene_ontology/go-basic.obo", go2items=go2geneids_human)

    # Find ALL GOs and GeneIDs associated with the function.

    gos = {go_term}
    # Add children GOs of the function GOs (note that all the `gos` variables are sets)
    gos_all = srchhelp.add_children_gos(gos)
    # Get Entrez GeneIDs for all the function GOs (w/ descendants)
    geneids = srchhelp.get_items(gos_all)
```

Thanks in advance!

**Comment by DylanSteinecke** [comment]
I'm also trying to do something similar. I am looking to see if I can compute the enrichment values for one particular term without computing all other terms. The best I have found in my brief search is restricting the ontology namespace to BP, MF, or CC--which I see you have done too. It does make the calculations proportionally quicker (e.g., takes about 1/3 the time when you are using 1 of 3 ontologies). The prt_study_gos_only doesn't appear to affect computing time, just what is printed after the computing.

**Comment by jasperhyp** [comment]
So if I remember correctly `AnnoOptions` actually does this.  It has options `ev_include` and also `keep_NOT`.  Then it can be input to a `gaf.reduce_annotations`.
